libraries/chain/include/eosio/chain/kv_config.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <fc/reflect/reflect.hpp>
+#include <cstdint>
+
+namespace eosio { namespace chain {
+
+   /**
+    * @brief limits for a kv database.
+    *
+    * Each database (ram or disk, currently) has its own limits for these parameters.
+    * The key and value limits apply when adding or modifying elements.  They may be reduced
+    * below existing database entries.
+    */
+   struct kv_database_config {
```

Comment:
It won't quite disallow usage.  You could still have an empty key with an empty value.

Reply:
Okay. Can you explicitly value initialize the fields of `kv_database_config` so that it is robust to modifications of the structs in ways that seem like they shouldn't change things but do in often subtle ways? (Example someone throwing an explicit default constructor on `kv_config` and forgetting to initialize `kvram` and `kvdisk` explicitly.)